scipy/stats/tests/test_stats.py

```diff
@@ -6615,6 +6615,24 @@ def test_mudholkar_george_equal_fisher_minus_pearson(self):
         # 0.5 here is because logistic = log(u) - log(1-u), i.e. no 2 factors
         assert_approx_equal(0.5 * (Z_f-Z_p), Z, significant=4)
 
+    @pytest.mark.parametrize("variant", ["single", "all"])
+    @pytest.mark.parametrize("method",
+                             ["fisher", "pearson", "tippett",
+                              "stouffer", "mudholkar_george"])
+    def test_monotony(self, variant, method):
+        # Test that result increases monotonously with respect to input.
+        changing_values = np.linspace(0.1, 0.9, 10)
+        rng = np.random.default_rng(278448169958891062669391462690811630763)
+        pvalues = rng.random(7)
+        combined_pvalues = []
+        for changing_value in changing_values:
+            if variant == "single":
+                pvalues[0] = changing_value
+            else:
+                pvalues = np.full(7, changing_value)
+            combined_p = stats.combine_pvalues(pvalues, method=method)[1]
+            combined_pvalues.append(combined_p)
+        assert np.all(np.diff(combined_pvalues) >= 0)
```

Comment:
I found the logic here a little confusing. For the "all" option, it's generating a set of p_values but then
immediately overwriting them with a constant array. It ends up just testing with an increasing sequence of constant arrays. What do you think of the following modification? It generates an increasing sequence of (non-constant) arrays all at once and then replaces the loop with a list comprehension. It leaves out the "single" option, but I'm not sure this option is actually necessary.

```suggestion
        m, n = 10, 7
        # Generate an m x n array of p_values. Each row corresponds to
        # a set of p_values to be combined with p_values increasing
        # monotonically down each column.
        p_values_array = np.sort(rng.uniform(0, 1, size=(m, n)), axis=0)
        # Generate combined p_values for each row in array.
        combined_pvalues = [
            stats.combine_pvalues(p_values_array[i, :], method=method)[1]
            for i in range(m)
        ]
        # Check that combined p_values increase monotonically as expected.
        assert np.all(np.diff(combined_pvalues) >= 0)
```

Reply:
I updated this as I described above. For each of the three variants the test data is created separately (and hopefully clearly).